community/io/src/test/java/org/neo4j/graphdb/mockfs/EphemeralFileSystemAbstraction.java

```diff
@@ -345,7 +345,7 @@ public boolean renameFile( File from, File to ) throws IOException
     @Override
     public File[] listFiles( File directory )
     {
-        if ( files.containsKey( directory ) )
+        if ( files.containsKey( directory ) || !directories.contains( directory ) )
```

Comment:
Could we not assume that `files` does not contain directories, and simplify this check to only `!directories.contains( directory )`?